### DISCUSSION TITLE
Lamellar microstructures with different width values of lames

### DISCUSSION THREAD

**Discussion by J-AKAZ** [neperfepx/neper#389] [Q&A] [answered]
Hello,
I am trying to create a lamellar microstructure and I wonder if we can assign different numerical values to the width of the lames. For example, we consider the microstructure presented in the figure below : 
![lamellar](https://user-images.githubusercontent.com/79088662/154937779-39374aa8-053e-4c06-85c6-c29c5495b42e.jpg)
I would like to know if we can give a width value to the lameswith id=1 different to the value attributed to the lames with id=2,  and even better if we can define a distribution law (lognormal, gaussian, ...) of the width of these lames ? If yes, could you tell me the command line to define on Neper ?

Thank you,

**Comment by rquey** [comment, marked as answer]
See option [-morpho](https://neper.info/doc/neper_t.html#cmdoption-morpho).

You can use `-morpho "lamellar(w=<val1>:<val2>)"` to specify 2 lamella widths (changing from one to the next, as on your image).

Example:
```
$ neper -T -n 10::from_morpho -morpho "voronoi::lamellar(w=0.1:0.05)" -o d
$ neper -V d.tess -datacellcol "scaleid(1)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/156526164-f1c72a4e-8a9a-454c-b219-b204f98c4021.png)

It is not possible to provide a distribution directly, but you can input more than 2 values:
```
$ string=`gsl-randist 1 100 gaussian 0.02 | awk '{val=$1+0.04; if (val>0) print val}' | sed -z 's/\n/:/g;s/:$/\n/'`
$ echo $string
0.046128:0.0315123:0.0359091:0.0314253:0.0661094:0.0529495:0.0638847:0.0051905:0.0452142:0.0346435:0.0337193:0.0406731:0.0522957:0.0161655:0.0001331:0.0520305:0.0527881:0.0279382:0.0436655:0.0471066:0.0439653:0.0758568:0.0380014:0.0119767:0.0178239:0.0234536:0.0608127:0.0338593:0.0535635:0.0231366:0.0315085:0.0269926:0.0104489:0.0511753:0.027038:0.0414627:0.0514716:0.0219271:0.0930161:0.0418659:0.078267:0.0092673:0.0196813:0.041592:0.0516273:0.0386133:0.0620483:0.0279074:0.0223743:0.0692245:0.0591681:0.0418422:0.0368299:0.0304906:0.0573398:0.0491867:0.0359693:0.0576594:0.0543901:0.0274083:0.0470161:0.0699526:0.0251657:0.0645097:0.0234426:0.0453814:0.0597885:0.0395107:0.0266487:0.044781:0.0529535:0.0052516:0.0553889:0.0437249:0.005248:0.0188598:0.0562378:0.0348443:0.0468765:0.0310513:0.0435657:0.0280541:0.0500047:0.0378323:0.0481913:0.033169:0.0410573:0.0246961:0.0432876:0.0385937:0.0388741:0.0402004:0.0318919:0.0313998:0.0090218:0.0497775:0.0301654:0.0333781:0.0304248
$ neper -T -n 10::from_morpho -morpho "voronoi::lamellar(w=$string)" -o d
$ neper --rcfile none -V d.tess -datacellcol "scaleid(1)" -print img
```
![image](https://user-images.githubusercontent.com/2659376/156529394-7acf4a2e-ce19-4449-9d4d-06544f0b753b.png)

**Comment by J-AKAZ** [comment]
Thank you for your answer. I have another question:
Is there any way to assign an id that equals the lame number where it belongs? I mean having an id that equals 1 or 2 (as shown in my image) instead of an id that is equal to the grain where it belongs?
I add that I want to have a biphasic medium where the lames with id=1 belong to one phase and the lames with id=2 belong to another phase.
I used the function -id 1::2 but it does not work for me in this case :
`neper -T -n 10::from_morpho -id 1::2 -morpho "voronoi::lamellar(w=0.1:0.05)" -o d -format geo,tess`
and i get in my d.tess : 
```
*id
   1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26
   27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49
   50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72
   73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95
   96 97 98 99 100 101 102 103 104 105 106 107 108 109 110
```
I want to have 
```
*id
  1 2 1 2 1 2 1 2 1 2 1 2 1 2 1 2 1....etc
```

Thank you so much

**Reply by rquey** [reply]
`id` correspond to a cell id and cannot be shared by different cells.

Use `-group lam` in `-T` to define groups (to represent material phases) corresponding to the 2 lamella widths.

You can know in which group cells are using `-statcell group`, and in which group elsets or elts are using `-statelset group` or `-statelt group`.
